todoapp/src/App.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import {ToggleLeft, ToggleRight} from 'react-feather'
+
+import {connect} from 'react-redux'
+import {addTodo, deleteTodo, toggleComplete} from './actions'
+import Todo from './components/Todo'
+
+class App extends Component {
+  constructor(props){
+    super(props)
+
+    this.state = {
+      newtodo: '',
+      showAll: false,
+      completedTodos: []
+    }
+  }
+
+  componentDidUpdate(prevProps){
+    if(prevProps.todos !== this.props.todos){
+      this.setState({
+        completedTodos: this.props.todos.filter(todo => todo.isCompleted)
+      })
+    }
+  }
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  handleToggleAll  = () => {
+    this.setState({
+      showAll: !this.state.showAll
+    })
+  }
+
+  handleSubmit = e => {
+    e.preventDefault()
+    if(this.state.newtodo === '') {
+      alert('what do you need to do today?')
+      return
+    }
+    this.setState({
+        newtodo: ''
+      })
+    this.props.addTodo(this.state.newtodo)
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <div>
+            <form onSubmit = {this.handleSubmit}>
+              <input placeholder = 'New Task' type='text' name='newtodo' value = {this.state.newtodo} onChange = {this.handleChange}/>
```

Comment:
The toggle works, but the toggle box on the left side is not functional unless you are in the deleted list.

Reply:
By default, it shows incomplete tasks, the big toggle is for showing all tasks. So if unchecked, the task would show up again when toggled back to view only incompletes.